### PR TITLE
MULE-8556: SpringRegistry is not thread-safe

### PR DESCRIPTION
**Pull request by marianogonzalez**

### CODE REVIEW

File: tests/integration/src/test/java/org/mule/config/spring/SpringRegistryConcurrenctyTestCase.java [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Test;
+
+public class SpringRegistryConcurrenctyTestCase extends FunctionalTestCase
```
Comment (aiannucci) [inline tests/integration/src/test/java/org/mule/config/spring/SpringRegistryConcurrenctyTestCase.java]: Class name is mispelled. Concurrencty -> Concurrency

File: tests/integration/src/test/java/org/mule/config/spring/SpringRegistryConcurrenctyTestCase.java [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+    private static final TimeUnit LATCH_TIME_UNIT = TimeUnit.SECONDS;
+
+    private CountDownLatch latch;
+    private List<Thread> threads;
+    private List<Map<String, Object>> reads;
+
+    @Override
+    protected String[] getConfigFiles()
+    {
+        return new String[] {};
```
Comment (pablokraan) [inline tests/integration/src/test/java/org/mule/config/spring/SpringRegistryConcurrenctyTestCase.java]: Why a FunctionalTestCase if there are no config files?

File: tests/integration/src/test/java/org/mule/config/spring/SpringRegistryConcurrenctyTestCase.java [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Test;
+
+public class SpringRegistryConcurrenctyTestCase extends FunctionalTestCase
+{
+
+    private static final int THREAD_COUNT = 10;
+    private static final long LATCH_TIMEOUT = 5;
+    private static final TimeUnit LATCH_TIME_UNIT = TimeUnit.SECONDS;
```
Comment (pablokraan) [inline tests/integration/src/test/java/org/mule/config/spring/SpringRegistryConcurrenctyTestCase.java]: Is this constant needed?

File: tests/integration/src/test/java/org/mule/config/spring/SpringRegistryConcurrenctyTestCase.java [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+        threads = new ArrayList<>(THREAD_COUNT);
+        reads = new ArrayList<>(THREAD_COUNT);
+    }
+
+    @Test
+    public void concurrentReads() throws Exception
+    {
+        final int registrySize = muleContext.getRegistry().lookupByType(Object.class).values().size();
+
+        for (int i = 0; i < THREAD_COUNT; i++)
```
Comment (pablokraan) [inline tests/integration/src/test/java/org/mule/config/spring/SpringRegistryConcurrenctyTestCase.java]: This kind of test could be flaky. What is needed is to test that one thread can read while another one is already reading and that when one thread is writing then any other thread will be blocked
https://github.com/mulesoft/mule/pull/1668/files#

Comment (marianogonzalez) [inline tests/integration/src/test/java/org/mule/config/spring/SpringRegistryConcurrenctyTestCase.java]: that's exactly what is tested here

File: tests/integration/src/test/java/org/mule/config/spring/SpringRegistryConcurrenctyTestCase.java (hunk 2 above)
Comment (marianogonzalez) [inline tests/integration/src/test/java/org/mule/config/spring/SpringRegistryConcurrenctyTestCase.java]: I need a started MuleContext with a backing SpringRegistry. This is the easiest way of achieving that. I could just use an AbstractMuleTestCase and initialise that on my own but then I'd be repeating code that FunctionalTestCase already has

**Pull request closed without merging**